### PR TITLE
snap: allow for different compressions for pack

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: schema/snapcraft.json [hunk 1]
```diff
@@ -412,6 +412,14 @@
             "description": "the snap epoch, used to specify upgrade paths",
             "format": "epoch"
         },
+        "compression": {
+            "description": "compression to use during packing",
+            "type": "string",
+            "enum": [
+                "lzo",
+                "xz"
+            ]
```
Comment (cjp256) [inline schema/snapcraft.json:421]: maybe set the default here?

Comment (cjp256) [reply schema/snapcraft.json:421]: so it's clear to the user what the default is...

Comment (sergiusens) [reply schema/snapcraft.json:421]: I want "snap pack" to choose what the default is, and maybe that varies depending on snap contents

**Comment by sergiusens** [comment]
boo, test failing as I retried the store part that got stuck and now have duplicate same snap comments.

File: schema/snapcraft.json [hunk 2]
```diff
@@ -412,6 +412,14 @@
             "description": "the snap epoch, used to specify upgrade paths",
             "format": "epoch"
         },
+        "compression": {
+            "description": "compression to use during packing",
```
Comment (sergiusens) [inline schema/snapcraft.json:416]:
```suggestion
            "description": "compression to use during packing, default if not set is determined by snap pack",
```

File: tests/spread/general/pack/task.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+summary: Different compression algorithms for snap pack
```
Comment (sergiusens) [inline tests/spread/general/pack/task.yaml:1]:
```suggestion
summary: Try different compression algorithms set in snapcraft.yaml
```

**Comment by sergiusens** [comment]
Closing in favor of #3189

**Pull request closed without merging**